CMakeLists.txt

```diff
@@ -238,6 +243,9 @@ elseif ("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
     set(CMAKE_C_FLAGS           "${CMAKE_C_FLAGS} /wd4701")
 endif()
 
+set(CMAKE_C_FLAGS_DEBUG     "${CMAKE_C_FLAGS_DEBUG} -O0")
+set(CMAKE_C_FLAGS_Release   "${CMAKE_C_FLAGS_Release} -O2")
```

Comment:
er... -O2 should have been be part of the compile flags already.  oops.  Let's set this, but on by default and for release builds.  If we need a non-optimized debug build, add something like "DebugNoOpt" to CMAKE_CONFIGURATION_TYPES that switches to -O0.
Also, per https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html, the variables end up all upper-case, i.e. CMAKE_C_FLAGS_RELEASE

Reply:
ha, I tested the debug ```-O0```/default non```-O0``` and found the effect is basically the same. They are both have no compilation optimization. So we do not need to set CMAKE_C_FLAGS_DEBUG -O0.